pmd-core/src/main/java/net/sourceforge/pmd/cli/PMDParameters.java

```diff
@@ -125,7 +126,20 @@ public Properties convert(String value) {
         }
     }
 
+
+    public static class RulePriorityValidator implements IValueValidator<Integer> {
+
+        @Override
+        public void validate(String name, Integer value) throws ParameterException {
+            if (value < 1 || value > 5) {
+                throw new ParameterException("Priority values can only be integer value, between 1 and 5," + value + " is not valid");
+            }
+        }
+    }
+
     // this has to be a public static class, so that JCommander can use it!
```

Comment:
I'll move this comment up to the new RulePriorityValidator class when I merge it